src/lpython/semantics/python_comptime_eval.h

```diff
@@ -211,13 +211,19 @@ struct PythonIntrinsicProcedures {
         if (ASRUtils::is_integer(*type)) {
             int64_t a = ASR::down_cast<ASR::IntegerConstant_t>(arg1)->m_n;
             int64_t b = ASR::down_cast<ASR::IntegerConstant_t>(arg2)->m_n;
+            if(b == 0){
+                throw SemanticError("Integer division or modulo by zero",loc);
+            }
             return ASR::down_cast<ASR::expr_t>(
-                ASR::make_IntegerConstant_t(al, loc, a%b, type));
+                ASR::make_IntegerConstant_t(al, loc, ((a%b)+b)%b, type));
```

Comment:
So you are trying to handle negative and positive combinations through this, correct? What about doing, `a > 0 && b > 0`, `a < 0 && b < 0`, `a > 0 && b < 0`, `a < 0 && b > 0` and `a == 0` in an if-else-if ladder. Right now the formula seems very cryptic. Or if you have a reference from where you have taken it to the formula then add it.

Reply:
I believe he used the formula from my comment here: https://github.com/lcompilers/lpython/issues/1562#issuecomment-1456821096
I dry ran different combinations of positive and negative values before writing my comment, and it seems to work fine.

Reply:
>.Or if you have a reference from where you have taken it to the formula then add it.

You can refer to this link, It correctly handles all the cases.
https://stackoverflow.com/questions/1907565/c-and-python-different-behaviour-of-the-modulo-operation

Reply:
Please add the above link in your code comments as well.

Reply:
I have added comments for my changes. You can have a look now .